test/test_dataflow.py

```diff
@@ -96,132 +130,126 @@ def test_should_run_beam_job(self, is_in_terminal_state_mock):
+        state_mock.return_value = 'RUNNING'
+
+        driver = CountWordsDriver()
+
         job = BeamJob(
             id='count_words',
-            entry_point=driver.run,
-            entry_point_arguments={
-                'words_to_filter': ['valid', 'word'],
-                'words_to_count': ['trash', 'valid', 'word', 'valid']
-            },
+            entry_point=driver.nope,
             test_pipeline=self._test_pipeline_with_label('count_words'),
             execution_timeout_sec=600)
 
-        count_words = Workflow(
-            workflow_id='count_words',
-            definition=[job])
-
         # then
         with self.assertRaises(RuntimeError) as e:
-            # when
-            count_words.run('2020-01-01')
+            job.execute(JobContext.make())
 
         # then
         self.assertEqual(cancel_mock.call_count, 1)
         wait_until_finish_mock.assert_called_with((600 - DEFAULT_PIPELINE_LEVEL_EXECUTION_TIMEOUT_SHIFT_IN_SECONDS) * 1000)
 
-    @patch.object(RunnerResult, 'is_in_terminal_state', create=True)
+    @patch.object(RunnerResult, 'state', new_callable=mock.PropertyMock)
     @patch.object(RunnerResult, 'cancel')
     @patch.object(RunnerResult, 'wait_until_finish')
-    def test_should_run_beam_job_with_timeout_without_cancel(self, wait_until_finish_mock, cancel_mock,
-                                                             is_in_terminal_state_mock):
+    def test_should_run_beam_job_with_timeout_without_cancel(
+        self,
+        wait_until_finish_mock,
+        cancel_mock,
+        state_mock,
+        locate_project_mock,
+    ):
         # given
         wait_until_finish_mock.return_value = 'DONE'
-        is_in_terminal_state_mock.return_value = True
-        driver = CountWordsDriver(CounterStater(CounterState()))
+        state_mock.return_value = 'DONE'
+
+        driver = CountWordsDriver()
         job = BeamJob(
             id='count_words',
-            entry_point=driver.run,
-            entry_point_arguments={
-                'words_to_filter': ['valid', 'word'],
-                'words_to_count': ['trash', 'valid', 'word', 'valid']
-            },
+            entry_point=driver.nope,
             test_pipeline=self._test_pipeline_with_label('count_words'),
             execution_timeout_sec=600)
 
-        count_words = Workflow(
-            workflow_id='count_words',
-            definition=[job])
-
         # when
-        count_words.run('2020-01-01')
+        job.execute(JobContext.make())
 
         # then
         self.assertEqual(cancel_mock.call_count, 0)
         wait_until_finish_mock.assert_called_with((600 - DEFAULT_PIPELINE_LEVEL_EXECUTION_TIMEOUT_SHIFT_IN_SECONDS) * 1000)
 
-    def test_should_throw_if_wait_until_finish_set_to_false_and_execution_timeout_passed(self):
+    def test_should_throw_if_wait_until_finish_set_to_false_and_execution_timeout_passed(
+        self,
+        locate_project_mock,
```

Comment:
Is it used?

Reply:
'locate_project_mock' is not used, but mock is still passed as argument by `unitetest.mock`

Reply:
ok